realm/realm-library/src/main/java/io/realm/RealmDictionary.java

```diff
@@ -100,28 +101,42 @@ public RealmDictionary(RealmDictionary<V> dictionary) {
     }
 
     private static <V> ManagedMapStrategy<String, V> getStrategy(Class<V> valueClass, BaseRealm baseRealm, OsMap osMap) {
+        if (isClassForRealmModel(valueClass)) {
+            ClassContainer classContainer = new ClassContainer(valueClass, null);
+            MapValueOperator<V> realmModelValueOperator = new RealmModelValueOperator<>(baseRealm, osMap, classContainer);
+            DictionaryManager<V> dictionaryManager = new DictionaryManager<>(realmModelValueOperator);
+            return new ManagedMapStrategy<>(dictionaryManager);
+        }
         return getStrategy(valueClass.getCanonicalName(), baseRealm, osMap);
     }
 
     private static <V> ManagedMapStrategy<String, V> getStrategy(String valueClass, BaseRealm baseRealm, OsMap osMap) {
-        DictionaryManager<V> operator = getOperator(valueClass, baseRealm, osMap);
-        return new ManagedMapStrategy<>(operator);
+        DictionaryManager<V> manager = getManager(valueClass, baseRealm, osMap);
+        return new ManagedMapStrategy<>(manager);
     }
 
-    private static <V> DictionaryManager<V> getOperator(String valueClass, BaseRealm baseRealm, OsMap osMap) {
+    private static <V> DictionaryManager<V> getManager(String valueClass, BaseRealm baseRealm, OsMap osMap) {
         // TODO: add other types when ready
         DictionaryManager<V> managedMapOperator;
         if (valueClass.equals(Mixed.class.getCanonicalName())) {
-            MapValueOperator<Mixed> mixedValueOperator = new MixedValueOperator(baseRealm, osMap);
+            MapValueOperator<Mixed> mixedValueOperator = new MixedValueOperator(baseRealm, osMap, new ClassContainer(null, valueClass));
             //noinspection unchecked
             managedMapOperator = new DictionaryManager<>((MapValueOperator<V>) mixedValueOperator);
         } else if (valueClass.equals(Boolean.class.getCanonicalName())) {
-            MapValueOperator<Boolean> mixedValueOperator = new BooleanValueOperator(baseRealm, osMap);
+            MapValueOperator<Boolean> mixedValueOperator = new BoxableValueOperator<>(baseRealm, osMap, new ClassContainer(null, valueClass));
+            //noinspection unchecked
+            managedMapOperator = new DictionaryManager<>((MapValueOperator<V>) mixedValueOperator);
+        } else if (valueClass.equals(UUID.class.getCanonicalName())) {
+            MapValueOperator<UUID> mixedValueOperator = new BoxableValueOperator<>(baseRealm, osMap, new ClassContainer(null, valueClass));
```

Comment:
```suggestion
            MapValueOperator<V> boxableValueOperator = new BoxableValueOperator<>(baseRealm, osMap, new ClassContainer(null, valueClass));
```
I believe that you can unify all cases except Mixed in one single case.

Reply:
Yeah, I tried, but the default typecasting to the generic fails when converting long to byte, short and integer. We need to cast it to the non-primitive first and then call "intValue", "shortValue" and "byteValue" - it can be seen much clearer in the next PR.